File: Data Structures & Algorithms/maximum-subarray/submission-1.java
class Solution {
    public int maxSubArray(int[] nums) {
        int res = -20000;
        int sum = -20000;
        for (int num: nums) {
            if (num > sum + num) sum = num;
            else sum += num;
            res = Math.max(res, sum);
        }
        return res;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = lo = hi = nums[0]
        
        for i in range(1, len(nums)):
            lo += nums[i]
            hi += nums[i]
            lo, hi = min(lo, hi, nums[i]), max(lo, hi, nums[i])
            res = max(res, hi)
        return res